packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -354,9 +354,11 @@ - (void)setupEventSinkIfReadyToPlay {
     }
 
     BOOL hasVideoTracks = [asset tracksWithMediaType:AVMediaTypeVideo].count != 0;
+    BOOL hasNoTracks = asset.tracks.count == 0;
 
-    // The player has not yet initialized when it contains video tracks.
-    if (hasVideoTracks && height == CGSizeZero.height && width == CGSizeZero.width) {
+    // The player has not yet initialized when it has no size, unless it is an audio-only track.
+    // HLS m3u8 video files never load any tracks, and are also not yet initialized until they have a size.
```

Comment:
Is it possible that any other formats would suffer from this?

Reply:
Possibly, we have limited testing for the range of possible video types.
At least now it will be no different than before https://github.com/flutter/plugins/pull/4639 in terms of file formats that don't have tracks.